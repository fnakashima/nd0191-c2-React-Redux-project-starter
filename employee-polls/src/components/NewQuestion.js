import { useState } from "react";
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { handleAddQuestion } from "../actions/questions";

const NewQuestion = ({ dispatch }) => {
  const navigate = useNavigate();
  const [optionOne, setOptionOne] = useState("");
  const [optionTwo, setOptionTwo] = useState("");

  const handleChange = (e) => {
    const { name, value } = e.target;
    //console.log(name, value);
    if (name === "optionOne") {
      setOptionOne(value);
    } else {
      setOptionTwo(value);
    }
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    //console.log("submit: ", optionOne, optionTwo);

    dispatch(handleAddQuestion(optionOne, optionTwo));

    setOptionOne("");
    setOptionTwo("");

    navigate("/");
  };

  return (
    <div className="new-poll">
      <h2>Would You Rather</h2>
      <label>Create Your Own Poll</label>
      <form onSubmit={handleSubmit}>
        <label>First Option</label>
        <input
          type="text"
          name="optionOne"
          placeholder="Enter Option One Text Here"
          onChange={handleChange}
        />
        <label>Second Option</label>
        <input
          type="text"
          name="optionTwo"
          placeholder="Enter Option Two Text Here"
          onChange={handleChange}
        />
        <button
          className="btn"
          type="submit"
          disabled={
            optionOne === "" ||
            optionOne.trim() === "" ||
            optionTwo === "" ||
            optionTwo.trim() === ""
          }
        >
          Submit
        </button>
      </form>
    </div>
  );
};

export default connect()(NewQuestion);
