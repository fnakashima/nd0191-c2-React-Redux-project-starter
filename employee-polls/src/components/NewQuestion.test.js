import * as React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import NewQuestion from "./NewQuestion";

const mockStore = configureStore();
const store = mockStore({});
// At least one unit test must use the render method from @testing-library/react to render one of your React components. The unit test should then perform some actions on the component using fireEvent such as fireEvent.click() or fireEvent.change(). After calling fireEvent, call the expect() method from jest to verify that a change occurred in the UI after the event was fired.
describe("NewQuestion", () => {
  it("should match snapshot", () => {
    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <NewQuestion />
        </MemoryRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  it("should update the input value when the user types", () => {
    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <NewQuestion />
        </MemoryRouter>
      </Provider>
    );

    const inputOne = component.getByTestId("option-one");
    const inputTwo = component.getByTestId("option-two");

    const inputOneText = "Option One";
    const inputTwoText = "Option Two";
    fireEvent.change(inputOne, { target: { value: inputOneText } });
    fireEvent.change(inputTwo, { target: { value: inputTwoText } });

    expect(inputOne.value).toEqual(inputOneText);
    expect(inputTwo.value).toEqual(inputTwoText);
  });
});
