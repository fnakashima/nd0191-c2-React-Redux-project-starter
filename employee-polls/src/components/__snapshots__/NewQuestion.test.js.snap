// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestion should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="new-poll"
      >
        <h2>
          Would You Rather
        </h2>
        <label>
          Create Your Own Poll
        </label>
        <form>
          <label>
            First Option
          </label>
          <input
            data-testid="option-one"
            name="optionOne"
            placeholder="Enter Option One Text Here"
            type="text"
          />
          <label>
            Second Option
          </label>
          <input
            data-testid="option-two"
            name="optionTwo"
            placeholder="Enter Option Two Text Here"
            type="text"
          />
          <button
            class="btn"
            disabled=""
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="new-poll"
    >
      <h2>
        Would You Rather
      </h2>
      <label>
        Create Your Own Poll
      </label>
      <form>
        <label>
          First Option
        </label>
        <input
          data-testid="option-one"
          name="optionOne"
          placeholder="Enter Option One Text Here"
          type="text"
        />
        <label>
          Second Option
        </label>
        <input
          data-testid="option-two"
          name="optionTwo"
          placeholder="Enter Option Two Text Here"
          type="text"
        />
        <button
          class="btn"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
